### PR TITLE
Fix picking perf when dragging

### PR DESCRIPTION
**Pull request by Pessimistress**
- Update EventManager
- Add `isDown` property to pointer events
- Prevent picking when dragging

### CODE REVIEW

File: src/utils/events/event-manager.js [hunk 1]
```diff
@@ -136,6 +142,8 @@ export default class EventManager {
    * Process the event deregistration for a single event + handler.
    */
   _removeEventHandler(event, handler) {
+    let success = false;
```
Comment (ericsoco) [inline src/utils/events/event-manager.js]: nit: call this `eventHandlerRemoved` to be a bit more explicit about what succeeded

File: src/utils/events/event-manager.js [hunk 2]
```diff
@@ -163,7 +186,6 @@ export default class EventManager {
       };
 
       // Calculate center relative to the root element
-      // TODO/xiaoji - avoid using getBoundingClientRect for perf?
```
Comment (ericsoco) [inline src/utils/events/event-manager.js]: Did you decide this is not a perf concern?

File: src/utils/events/event-manager.js [hunk 3]
```diff
@@ -191,7 +213,7 @@ export default class EventManager {
     const alias = BASIC_EVENT_ALIASES[srcEvent.type];
     if (alias) {
       // fire all events aliased to srcEvent.type
-      const emitEvent = Object.assign({}, event, {type: alias});
+      const emitEvent = Object.assign({}, event, {isDown: true, type: alias});
```
Comment (ericsoco) [inline src/utils/events/event-manager.js]: Looks like you're always setting `isDown: true` here, should the third `Object.assign()` param be the second?

File: src/utils/events/event-manager.js (hunk 2 above)
Comment (Pessimistress) [inline src/utils/events/event-manager.js]: I don't find it prominent during profiling.

File: src/utils/events/event-manager.js (hunk 3 above)
Comment (Pessimistress) [inline src/utils/events/event-manager.js]: we're overriding `event.type`.

**Comment by ericsoco** [comment]
Worth noting that this PR fixes the more obvious perf problem during dragging (perf slows down the screen update while dragging), but not the more subtle perf problem that happens when running picking on hover (mousemove without mousedown). The latter is for a future PR.

File: src/utils/events/event-manager.js (hunk 2 above)
Comment (ibgreen) [inline src/utils/events/event-manager.js]: May not be conclusive. It reportedly depends on the surrounding DOM, how deep we are in the DOM tree, styling etc.

Whether it is worth the trouble to avoid I don't know...

File: src/utils/events/event-manager.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import WheelInput from './wheel-input';
 import MoveInput from './move-input';
-import {isBrowser} from '../globals';
+import {isBrowser} from '../../lib/utils/globals';
```
Comment (ibgreen) [inline src/utils/events/event-manager.js]: Some of these utils should be moved up a level if they are no longer used only inside lib.

File: src/lib/layer-manager.js [hunk 5]
```diff
@@ -594,6 +594,10 @@ export default class LayerManager {
    *                        {Object} srcEvent:             native JS Event object
    */
   _onPointerMove(event) {
+    if (event.isDown) {
+      // Is dragging
```
Comment (ibgreen) [inline src/lib/layer-manager.js]: Can we add a little more to the comment  i.e. why does dragging mean that we don't need move events?

File: src/utils/globals.js [hunk 6]
```diff
@@ -1,15 +0,0 @@
-// Purpose: include this in your module to avoids adding dependencies on
```
Comment (ibgreen) [inline src/utils/globals.js]: Why was this deleted?

File: src/lib/layer-manager.js (hunk 5 above)
Comment (shaojingli) [inline src/lib/layer-manager.js]: this comment means we should not fire pointermove event when mouse is down because pointermove leads to pickLayer getting called. Yes, we probably want to explain a bit more in the comments, or considering put a link to this PR

Comment (ericsoco) [inline src/lib/layer-manager.js]: Well, more specifically `_onPointerMove` feeds into our hover handlers (`onHover` and `onLayerHover`), and dragging is not supported by the framework itself, it's the app's responsibility. But yeah, fleshing out the comment will help future detective work.

**Comment by Pessimistress** [comment]
https://github.com/uber/deck.gl/issues/769
